File: 128-longest-consecutive-sequence/longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
      Set<Integer> s = new HashSet<>();

      for(int n : nums){
        s.add(n);
      }   

      int longest = 0;
      for(int n : nums){
        if(s.contains(n-1)) continue;
        int j = 1;

        while(s.contains(n+j)) j++;
        longest = Math.max(longest , j);
      }

      return longest;
    }
}

File: 128-longest-consecutive-sequence/longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        s = set()

        for num in nums:
            s.add(num)

        length = 0

        for num in s:
            if num-1 in s:
                continue
            
            j = 1
            while num+j in s:
                j += 1
            
            length = max(length , j)
        
        return length